enhancements/network/user-defined-network-segmentation.md

```diff
@@ -413,6 +669,220 @@ guarantee uniqueness and eliminates the ability to falsify access.
+metadata:
+  name: db-network
+  namespace: demo
+  finalizers:
+  - k8s.ovn.org/user-defined-network-protection
+  ownerReferences:
+    - apiVersion: k8s.ovn.org/v1alpha1
+      blockOwnerDeletion: true
+      kind: UserDefinedNetwork
+      name: db-network
+      uid: f45efb13-9511-48c1-95d7-44ee17c949f4
+spec:
+  config: >    
+      '{
+          "cniVersion":"0.3.1",
+          "excludeSubnets":"10.0.0.100/24",
+          "mtu":1500,   
+          "name":"demo-network",  <--- generated unique network-name
+          "netAttachDefName":"demo/poc-db-network",
+          "subnets":"10.0.0.0/24",
+          "topology":"layer2",
+          "type":"ovn-k8s-cni-overlay",
+          "role": "primary",
+          "persistentIPs": "true"
+      }'
+```
+
+For cluster-scoped CRDs, the NAD `metadata.name` should correspond to the request CRD `metadata.name` with an additional prefix.
+Having the prefix avoids conflicting with existing NADs who has the same `metadata.name`.
+
+Creating cluster scoped CRD instance should trigger creation of the corresponding NAD at each namespace specified in the spec.
+Following the above cluster-scope CRD [example](#example---cluster-scoped-network), the following NADs should be created:
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: shared-db-network          <--- name starts with "shared"
+  namespace: mynamespace
+  finalizers:
+  - k8s.ovn.org/user-defined-network-protection
+  ownerReferences:
+    - apiVersion: k8s.ovn.org/v1alpha1
+      blockOwnerDeletion: true
+      kind: ClusterUserDefinedNetwork
+      name: db-network
+      uid: f45efb13-9511-48c1-95d7-44ee17c949f4 
+spec:
+  config: >    
+      '{
+          "cniVersion":"0.3.1",
+          "excludeSubnets":"10.0.0.100/24",
+          "mtu":1500,   
+          "name":"shared-db-network",   <--- generated unique network-name
+          "netAttachDefName":"mynamespace/db-network",
+          "subnets":"10.0.0.0/24",
+          "topology":"layer2",
+          "type":"ovn-k8s-cni-overlay",
+          "role": "primary"
+      }'
+---
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: shared-db-network                 <--- same name as in other nameapces
+  namespace: theirnamespace
+  finalizers:
+  - k8s.ovn.org/user-defined-network-protection
+  ownerReferences:
+  - apiVersion: k8s.ovn.org/v1alpha1
+    blockOwnerDeletion: true
+    kind: ClusterUserDefinedNetwork
+    name: db-network
+    uid: f45efb13-9511-48c1-95d7-44ee17c949f4
+spec:
+  config: >
+    '{
+        "cniVersion":"0.3.1",
+        "excludeSubnets":"10.0.0.100/24",
+        "mtu":1500,   
+        "name":"shared-db-network",   <--- same name as in other namespaces
+        "netAttachDefName":"theirnamespace/db-network",
+        "subnets":"10.0.0.0/24",
+        "topology":"layer2",
+        "type":"ovn-k8s-cni-overlay",
+        "role": "primary"
+    }'
+```
+
+##### Validations
```

Comment:
we may also want to validate minimum subnet sizes

Reply:
There are some checks like is this a valid ipv4/ipv6 subnet. I guess you will cover that with your regexp. There is also other validation that could be done, like making sure the host mask is > subnet mask. OVNK already checks for this here for the global subnet values:
https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/config/utils.go#L99

We can check for this in either OVNK or using CRD validation. Whatever is easier. There are also some CRD validation examples here:
https://github.com/openshift/api/blob/master/operator/v1/types_network.go#L486

im not sure they will work when we are using a format that has the host mask as well. Worst case just validate this stuff in OVNK and report back error status to the CR.